backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -285,27 +282,19 @@ public AddTestResultResponse addTestResult(
                   patientLinkId,
                   "Your Covid-19 test result is ready to view: " + patientLinkUrl + patientLinkId);
 
-          boolean hasDeliveryFailure =
+          boolean failure =
               smsSendResults.stream().anyMatch(delivery -> !delivery.getDeliverySuccess());
-
-          if (hasDeliveryFailure) {
-            return new AddTestResultResponse(savedOrder, false);
-          }
+          if (failure) deliveryStatuses.add(false);
         }
 
         if (emailDeliveryPreference(savedOrder) || smsAndEmailDeliveryPreference(savedOrder)) {
-          try {
-            testResultsDeliveryService.emailTestResults(patientLink);
-          } catch (IOException e) {
-            deliveryStatus = false;
-            log.error(
-                "failed to send email for patient link {}, exception: {}",
-                patientId,
-                e.getMessage());
-          }
+          boolean emailDeliveryStatus = testResultsDeliveryService.emailTestResults(patientLink);
+          deliveryStatuses.add(emailDeliveryStatus);
         }
       }
 
+      boolean deliveryStatus =
+          deliveryStatuses.isEmpty() || deliveryStatuses.stream().anyMatch(status -> status);
```

Comment:
IIUC this will return `true` if any of the delivery statuses are `true`. Is that the intent? In the logic this is replacing, the delivery status of the operation as a whole will be `false` if any individual deliveries failed.

Reply:
yeah, I was a bit torn on that and decided if they get an email or text, that the delivery status is good since that's gonna drive an error message in the UI. I'll add a ticket to break that return status from the api into more granular (email/sms) versus one boolean for both.

Reply:
The logic is actually replacing not sending an email if text failed (when they have ALL selected) and I dont think that was correct.